vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+names(obsData) <- lapply(obsData, function(x) x$name)
+```
+
+Typically, `DataCombined` is used to create figures using the `plotXXX()` functions. 
+The different plot types are described in [Visualizations with `DataCombined`](data-combined-plotting.html). 
+To visualize the different functionalities of `DataCombined` in this document, 
+we will use the `plotIndividualTimeProfile()` function.
+
+## Creating `DataCombined` object
+
+First, we create a new instance of `DataCombined` class.
+
+```{r}
+myDataCombined <- DataCombined$new()
+```
+
+Then we add simulated results to this object using `$addSimulationResults()`:
+
+```{r}
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma",
+  groups = "Aciclovir PVB"
+)
+```
+
+Next we add observed data to this object using `$addDataSets()`:
+
+```{r}
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`, 
+  groups = "Aciclovir PVB"
+)
+```
+
+Every data, be it simulated results or from `DataSet`, must have a unique name 
+within `DataCombined`. If not specified by user, the path of simulated results 
+or the `$name` property of the `DataSet` are used as the name. Alternatively, we 
+can define the name when adding the data, as in the above example adding simulated 
+results.
+
+There are a few things to keep in mind here:
+
+- It doesn't matter in which order observed or simulated datasets are added.
+- You can add multiple `DataSet` objects in `$addDataSets()` method call.
+- You can add only a single instance of `SimulationResults` in `$addSimulationResults()` method call.
+- If you add a dataset with the same name as existing (e.g., adding simulation results 
+from multiple simulations and not specifying the name, so the path of the results 
+is used as the name), the new data will replace the existing.
+
+## Grouping
+
+Since the `DataCombined` object can store many datasets, some of them may naturally form a grouping and you would wish to track this in the object. Both `$addDataSets()` and `$addSimulationResults()` allow group specification via `groups` argument.
+
+When being plotted, data sets without a grouping will appear with distinct color, 
+line style (for simulated results), or symbol style (for observed data) with a 
+separate entry in the legend, with the name of the data set being the legend entry.
+
+Let's create a `DataCombined` and add simulation results and observed data *without* 
+specifying their grouping and create a time profile:
+
+```{r}
+myDataCombined <- DataCombined$new()
+
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma"
+)
+
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`
+)
+
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+If you do not specify `groups` when you add datasets or set it to `NULL`, and 
```

Comment:
...`or set it to `NULL``...
Would remove this. Why should I set groups to NULL if I can just omit the argument?